core/uploadDiveLogsDE.cpp

```diff
@@ -40,15 +41,13 @@ void uploadDiveLogsDE::doUpload(bool selected, const QString &userid, const QStr
 
 
 	/* generate a temporary filename and create/open that file with zip_open */
-	QString filename(QDir::tempPath() + "/divelogsde-upload.dld");
-
-	// delete file if it exist
-
-	QFile f(filename);
-	if (f.open(QIODevice::ReadOnly)) {
-		f.close();
-		f.remove();
-	}
+	auto *tmpfile = new QTemporaryFile();
```

Comment:
Please no plain `new`! We have so frustratingly many leaks owing to this archaic lifetime management. The simplest way would be to do this in a separate function, as such:
```
static QString make_unique_filename()
{
        QTemporaryFile tmpfile;
        tmpfile->setFileTemplate(QDir::tempPath() + "/divelogsde-upload.XXXXXXXX.dld");
        return tmpfile->fileName();
}
```

Reply:
Done